src/Neo/Network/P2P/Payloads/Signer.cs

```diff
@@ -192,14 +192,14 @@ void IInteroperable.FromStackItem(VM.Types.StackItem stackItem)
 
         VM.Types.StackItem IInteroperable.ToStackItem(ReferenceCounter referenceCounter)
         {
-            return new VM.Types.Array(referenceCounter, new VM.Types.StackItem[]
-            {
+            return new VM.Types.Array(referenceCounter,
+            [
                 Account.ToArray(),
                 (byte)Scopes,
-                new VM.Types.Array(referenceCounter, AllowedContracts.Select(u => new VM.Types.ByteString(u.ToArray()))),
-                new VM.Types.Array(referenceCounter, AllowedGroups.Select(u => new VM.Types.ByteString(u.ToArray()))),
-                new VM.Types.Array(referenceCounter, Rules.Select(u => u.ToStackItem(referenceCounter)))
-            });
+                Scopes.HasFlag(WitnessScope.CustomContracts) ? new VM.Types.Array(referenceCounter, AllowedContracts.Select(u => new VM.Types.ByteString(u.ToArray()))) : new VM.Types.Array(referenceCounter),
+                Scopes.HasFlag(WitnessScope.CustomGroups) ? new VM.Types.Array(referenceCounter, AllowedGroups.Select(u => new VM.Types.ByteString(u.ToArray()))) : new VM.Types.Array(referenceCounter),
+                Scopes.HasFlag(WitnessScope.WitnessRules) ? new VM.Types.Array(referenceCounter, Rules.Select(u => u.ToStackItem(referenceCounter))) : new VM.Types.Array(referenceCounter)
+            ]);
```

Comment:
```suggestion
                AllowedContracts != null ? new VM.Types.Array(referenceCounter, AllowedContracts.Select(u => new VM.Types.ByteString(u.ToArray()))) : new VM.Types.Array(referenceCounter),
                AllowedGroups != null ? new VM.Types.Array(referenceCounter, AllowedGroups.Select(u => new VM.Types.ByteString(u.ToArray()))) : new VM.Types.Array(referenceCounter),
                Rules != null ? new VM.Types.Array(referenceCounter, Rules.Select(u => u.ToStackItem(referenceCounter))) : new VM.Types.Array(referenceCounter)
            ]);
```

The crash is happening because `AllowedContracts`, `AllowedGroups` and `Rules` is `null`, and doing `.Select()` will still 
cause a crash with those flags set. Take in count `ApplicationEngine.Run` and `Dynamic Scripts`.

Reply:
@shargon @roman-khimov @AnnaShaleva
This should work now.

Reply:
The flags are used in other serializable methods, so we should fix the flags if are wrong in cli

Reply:
But look at `Call Stack`->`NullException`

```
at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.Select[TSource,TResult](IEnumerable`1 source, Func`2 selector)
   at Neo.Network.P2P.Payloads.Signer.Neo.SmartContract.IInteroperable.ToStackItem(ReferenceCounter referenceCounter)
   at Neo.SmartContract.ApplicationEngine.Convert(Object value)
   at Neo.SmartContract.ApplicationEngine.<Convert>b__95_0(Object p)
```